apps/native/app/src/lib/useBrowser.ts

```diff
@@ -0,0 +1,51 @@
+import { Passkey } from 'react-native-passkey'
+import { authStore } from '../stores/auth-store'
+import { useFeatureFlag } from '../contexts/feature-flag-provider'
+import { preferencesStore } from '../stores/preferences-store'
+import { openNativeBrowser } from './rn-island'
+import { navigateTo } from './deep-linking'
+import { useAuthenticatePasskey } from './passkeys/useAuthenticatePasskey'
+import { addPasskeyAsLoginHint } from './passkeys/helpers'
+
+export const useBrowser = () => {
+  const { authenticatePasskey } = useAuthenticatePasskey()
+  const isPasskeyEnabled = useFeatureFlag('isPasskeyEnabled', false)
+
+  const openBrowser = async (url: string, componentId?: string) => {
+    const passkeysSupported: boolean = Passkey.isSupported()
+
+    const { hasOnboardedPasskeys, hasCreatedPasskey } =
+      preferencesStore.getState()
+
+    // If url includes minarsidur or umsoknir we need authentication so we check for passkeys
+    if (
+      passkeysSupported &&
+      isPasskeyEnabled &&
+      (url.includes('/minarsidur') || url.includes('/umsoknir'))
+    ) {
+      if (hasCreatedPasskey) {
+        // Don't show lockscreen behind native passkey modals
+        authStore.setState({
+          noLockScreenUntilNextAppStateActive: true,
+        })
+        // Open passkey flow to authenticate
+        const passkey = await authenticatePasskey()
+        if (passkey) {
+          const urlWithLoginHint = addPasskeyAsLoginHint(url, passkey)
+          urlWithLoginHint && openNativeBrowser(urlWithLoginHint, componentId)
+        }
+        // If something goes wrong we fail silently and open the browser normally
+        openNativeBrowser(url, componentId)
```

Comment:
Looks like if there's a passkey authentications this function runs `openNativeBrowser` twice.